datajunction-server/datajunction_server/construction/build_v2.py

```diff
@@ -217,6 +217,9 @@ async def get_measures_query(  # pylint: disable=too-many-locals
                 measure_columns.append(expr)
                 expr.set_semantic_type(SemanticType.MEASURE)  # type: ignore
         await parent_ast.compile(context)
+        dependencies, _ = await parent_ast.extract_dependencies(
```

Comment:
By extracting dependencies at this stage, we avoid calling another `query.compile()` call on the final query, which is expensive and without much benefit (all it does is add type inference).

Reply:
That one fallout of imprecise function names. Every time I see `build` or `compile` I need to go back to the source and remind myself what it means.

Reply:
Agreed... `compile` is kind of like a "hydrate query with metadata" stage, and `build` actually changes the structure of the query. If you can come up with a better function name that represents the concept happy to rename compile! 😅